src/transport/SessionManager.h

```diff
@@ -349,6 +349,8 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate, public FabricTabl
 
     void ExpireAllPASESessions();
 
+    bool MarkSessionAsDefunctMark(const ScopedNodeId & node, const Optional<Transport::SecureSession::Type> & type);
```

Comment:
This needs to be documented.  What does passing NullOptional for the second arg mean, for example?  What happens if there are multiple sessions that match the ScopedNodeId?

Reply:
Can there be multiple matches. I was under the impression there can only be one in the session manager

Reply:
I think we can end up with more than one.  Say node A establishes a session to node B.  Node B tries to send a message to A on the session, the message fails to be delivered, node B marks the session defunct on its end.  Node B tries to send another message to A, sets up a new session.

Now on B there are two sessions (one active, one defunct) and on A there are two sessions, both active.  Note how `SessionManager::FindSecureSessionForNode` handles this case by returning the most-recently-active of the set of active sessions....

Reply:
Okay so we are going to mark the most recently active session as defunct. I could be convinced that we should be marking them all as defunct, but all the call sites currently have been marking just one as defunct, although it technically might not be the most recently active session, in most cases when both locations that call into `MarkSessionAsDefunct` they were calling `FindExistingSessionSetup` it was the most recently active that would have been used by the applications are are calling the call site.

There is also a chance that it will be removed in [20452](https://github.com/project-chip/connectedhomeip/issues/20452)

Reply:
See my comment in OTA requestor, and why we need to fix this to actually mark all matching sessions as defunct, since it would otherwise be a regression for OTA.

Reply:
NVM Jerry recommends that we mark them all as defunct. In a follow up I will work on removing `MarkSessionsAsDefunct`